src/Bicep.LangServer/Handlers/BicepSignatureHelpHandler.cs

```diff
@@ -0,0 +1,271 @@
+        {
+            // local function
+            static Task<SignatureHelp?> NoHelp() => Task.FromResult<SignatureHelp?>(null);
+
+            CompilationContext? context = this.compilationManager.GetCompilation(request.TextDocument.Uri);
+            if (context == null)
+            {
+                return NoHelp();
+            }
+
+            int offset = PositionHelper.GetOffset(context.LineStarts, request.Position);
+            
+            var functionCall = GetActiveFunctionCall(context.ProgramSyntax, offset, out var arguments);
+            if (functionCall == null)
+            {
+                return NoHelp();
+            }
+
+            var semanticModel = context.Compilation.GetEntrypointSemanticModel();
+            var symbol = semanticModel.GetSymbolInfo(functionCall);
+            if (symbol is not FunctionSymbol functionSymbol)
+            {
+                // no symbol or symbol is not a function
+                return NoHelp();
+            }
+
+            // suppress ErrorType in arguments because the code is being written
+            // this prevents function signature mismatches due to errors
+            var normalizedArgumentTypes = NormalizeArgumentTypes(arguments, semanticModel);
+
+            var signatureHelp = CreateSignatureHelp(arguments, normalizedArgumentTypes, functionSymbol, offset);
+            TryReuseActiveSignature(request.Context, signatureHelp);
+
+            return Task.FromResult<SignatureHelp?>(signatureHelp);
+        }
+
+        private static SyntaxBase? GetActiveFunctionCall(ProgramSyntax syntax, int offset, out ImmutableArray<FunctionArgumentSyntax> arguments)
+        {
+            // if the cursor is placed after the closing paren of a function, it needs to count as outside of that function call
+            // for purposes of signature help (otherwise we'll show the wrong function when function calls are nested)
+            var matchingNodes = SyntaxMatcher.FindNodesMatchingOffsetExclusive(syntax, offset);
+
+            var index = matchingNodes
+                .FindLastIndex(
+                    matchingNodes.Count - 1,
+                    current => current is FunctionCallSyntax functionCall && TextSpan.BetweenExclusive(functionCall.OpenParen.Span, functionCall.CloseParen).ContainsInclusive(offset) ||
+                               current is InstanceFunctionCallSyntax instanceFunctionCall && TextSpan.BetweenExclusive(instanceFunctionCall.OpenParen.Span, instanceFunctionCall.CloseParen.Span).ContainsInclusive(offset));
+
+            var activeFunctionCall = index < 0 ? null : matchingNodes[index];
+            switch (activeFunctionCall)
+            {
+                case FunctionCallSyntax functionCall:
+                    arguments = functionCall.Arguments;
+                    return functionCall;
+
+                case InstanceFunctionCallSyntax instanceFunctionCall:
```

Comment:
>InstanceFunctionCallSyntax [](start = 21, length = 26)

The refactor to add a common base for functions and instance functions will simplify this code.

Reply:
This is done now.